polygon_inference.py

```diff
@@ -711,28 +812,42 @@ def _merge_polygons(
         # Find contours in the bitmap
         contours, _ = cv2.findContours(bitmap, cv2.RETR_EXTERNAL, cv2.CHAIN_APPROX_SIMPLE)
         
-        merged_polygons: List[np.ndarray] = []
+        # Collect all valid contours into shapely polygons
+        shapely_polygons = []
         
         for contour in contours:
             # Skip very small contours (area is scaled by scale_factor^2)
             area = cv2.contourArea(contour)
             if area < CFG.MIN_POLYGON_AREA * (scale_factor ** 2):
                 continue
             
-            # Simplify the contour to reduce jaggedness while preserving shape
-            perimeter = cv2.arcLength(contour, True)
-            epsilon = 0.01 * perimeter  # 1% of perimeter
-            simplified_contour = cv2.approxPolyDP(contour, epsilon, True)
+            # Convert contour to Shapely Polygon
+            contour_points = contour.reshape(-1, 2).astype(np.float64)
+            shapely_polygon = Polygon(contour_points)
+            shapely_polygons.append(shapely_polygon)
+        
+        # Create single GeoDataFrame with all polygons and regularize them all at once
+        if shapely_polygons:
+            gdf = gpd.GeoDataFrame({'geometry': shapely_polygons})
+            regularized_gdf = regularize_geodataframe(gdf, simplify_tolerance=20, parallel_threshold=100)
```

Comment:
The above change makes messier polygons, but this library does a good job of simplifying them.